docs/extensibility/snippets/MailDevResource/MailKit.Client/MailKitClientSettings.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Net;
+
+namespace MailKit.Client;
+
+/// <summary>
+/// Provides the client configuration settings for connecting MailKit to an SMTP server.
+/// </summary>
+public sealed class MailKitClientSettings
+{
+    /// <summary>
+    /// Gets or sets the SMTP server connection string. It should be represented in a
+    /// valid <see cref="Uri"/> form.
+    /// </summary>
+    /// <value>
+    /// The default value is <see langword="null"/>.
+    /// </value>
+    public string? ConnectionString { get; set; }
```

Comment:
Should this be `Uri? Endpoint` (or similar name)?

Reply:
Yeah, I like that better. But how would that change the existing code in the registration extensions, where we assign the given connection string from the call to `GetConnectionString`. And that impacts the deferred validation of the connection string parsing to a `Uri` too, so what would need to be updated there?